src/resolve/resolved-dns-stub.c

```diff
@@ -479,7 +493,7 @@ static int manager_dns_stub_udp_fd_extra(Manager *m, DNSStubListenerExtra *l) {
         if (r < 0)
                 goto fail;
 
-        if (bind(fd, &l->address.sockaddr.sa, l->address.size) < 0) {
+        if (bind(fd, &sa.sa, l->family == AF_INET ? sizeof(struct sockaddr_in) : sizeof(struct sockaddr_in6)) < 0) {
```

Comment:
maybe we should add a helper for this one day: 

SOCKADDR_LEN() which checks the family and then returns sizeof(sockaddr_in), sizeof(sockaddr_in6) or SOCKADDR_UN_LEN() or SOCKADDR_LL_LEN() or -EAFNOSUPPORT or so.